test/Chainweb/Test/Mempool/InMem.hs

```diff
@@ -20,16 +20,17 @@ tests = testGroup "Chainweb.Mempool.InMem"
             $ Chainweb.Test.Mempool.tests
             $ MempoolWithFunc wf
   where
+    wf :: (InsertCheck -> MempoolBackend MockTx -> IO a) -> IO a
     wf f = do
-        mv <- newMVar (\v -> return $! V.map (const Nothing) v)
-        let cfg = InMemConfig txcfg mockBlockGasLimit 2048 (checkMv mv)
+        mv <- newMVar (pure . V.map Right)
+        let cfg = InMemConfig txcfg mockBlockGasLimit 2048 Right (checkMv mv)
```

Comment:
Here and elsewhere I'm concerned that the one-off validations are no longer checked, and are reported as passing regardless of their contents.

Reply:
a) we test the low-level check code by populating the `MVar` with a check
b) some specific checks on chainweb transactions (`Mempool` is intentionally abstract in the transaction type for mock/testability) are performed in the remote pact tests -- but we need to make sure we cover this handler thoroughly